src/utils/test/utils.cpp

```diff
@@ -443,6 +443,26 @@ TEST(core, dlink)
     EXPECT_TRUE(count == 0);
 }
 
+TEST(core, find_string_prefix)
+{
+    struct test_case
+    {
+        std::string input;
+        char separator;
+        std::string expected_prefix;
+    } tests[] = {{"", ' ', ""},
```

Comment:
Could add cases that the separator is just the whole source string, and the source string only include separators, such as:
```c++
{{" ", ' ', ""}},
{{"..", '.', ""}},
```